network/src/client.rs

```diff
@@ -0,0 +1,24 @@
+use crate::constants;
+use std::{
+    convert::{TryFrom, TryInto},
+    error::Error,
+    io::prelude::*,
+    net::TcpStream,
+};
+
+/// Sends a `request` to server over `stream` and return response from server.
+pub fn send<I, O>(request: I, stream: &mut TcpStream) -> Result<O, Box<dyn Error>>
```

Comment:
It will be great to do not require `TcpStream` usage for clients. API should be agnostic.